src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -140,26 +142,45 @@ protected override async Task OnSave(Uri resourceUri, StorageContent content, Ca
                     }
 
                     destinationStream.Seek(0, SeekOrigin.Begin);
+
                     await blob.UploadFromStreamAsync(destinationStream,
-                                                     accessCondition: null,
-                                                     options: _blobRequestOptions,
-                                                     operationContext: null,
-                                                     cancellationToken: cancellationToken);
-                    Trace.WriteLine(String.Format("Saved compressed blob {0} to container {1}", blob.Uri.ToString(), _directory.Container.Name));
+                        accessCondition: null,
+                        options: _blobRequestOptions,
+                        operationContext: null,
+                        cancellationToken: cancellationToken);
+
+                    Trace.WriteLine(string.Format("Saved compressed blob {0} to container {1}", blob.Uri.ToString(), _directory.Container.Name));
                 }
             }
             else
             {
-                using (Stream stream = content.GetContentStream())
+                var copiableContent = content as AzureServerSideCopiableContent;
+
+                if (copiableContent == null)
+                {
+                    using (Stream stream = content.GetContentStream())
+                    {
+                        await blob.UploadFromStreamAsync(stream,
+                            accessCondition: null,
+                            options: _blobRequestOptions,
+                            operationContext: null,
+                            cancellationToken: cancellationToken);
+                    }
+                }
+                else
                 {
-                    await blob.UploadFromStreamAsync(stream,
-                                                     accessCondition: null,
-                                                     options: _blobRequestOptions,
-                                                     operationContext: null,
-                                                     cancellationToken: cancellationToken);
-                    Trace.WriteLine(String.Format("Saved uncompressed blob {0} to container {1}", blob.Uri.ToString(), _directory.Container.Name));
+                    await TransferManager.CopyAsync(copiableContent.SourceBlob, blob, isServiceCopy: true);
+
+                    // Copying (above) did not apply these properties so apply them again.
+                    blob.Properties.ContentType = content.ContentType;
```

Comment:
do we need to apply all properties? (future proofing)

Reply:
No, the only properties we set today are ContentType and CacheControl.